--- magnus-20060324.orig/configure
+++ magnus-20060324/configure
@@ -17,14 +17,15 @@
 #echo "Setting up default Magnus paths..."
 #echo
 
-DEFAULT_MAGNUSHOME=`pwd`
+DEFAULT_MAGNUSHOME=/usr/share/magnus
 DEFAULT_MAGNUSCURRENTHOME=`pwd`
 DEFAULT_WISH=/usr/bin/wish
 DEFAULT_MAGNUSTMP=/tmp
 MAGNUSTMP=/tmp
 echo "Which operation system are you going to use ? "
 echo "(Enter 0 for LINUX, 1 for System V, 2 for BSD 4.11+/Mac OS X; LINUX is default)"
-read OS
+# read OS
+OS=0
 
 if [ "$OS" = "1" ]
 then
@@ -44,7 +45,8 @@
 echo "Enter the path to the Magnus source tree (the directory where"
 echo "the Magnus source is currently located):"
 echo "[$DEFAULT_MAGNUSCURRENTHOME]"
-read MAGNUSCURRENTHOME
+# read MAGNUSCURRENTHOME
+MAGNUSCURRENTHOME=
 if [ "$MAGNUSCURRENTHOME" = "" ]
 then
 	MAGNUSCURRENTHOME=$DEFAULT_MAGNUSCURRENTHOME
@@ -94,12 +96,14 @@
 
 # in the meantime, just do this
 
-MAGNUSHOME=$MAGNUSCURRENTHOME
+# MAGNUSHOME=$MAGNUSCURRENTHOME
+MAGNUSHOME=$DEFAULT_MAGNUSHOME
 
 echo
 echo "Enter the path to the Wish executable (with the addinput extensions):"
 echo "[$DEFAULT_WISH]"
-read WISH
+# read WISH
+WISH=
 if [ "$WISH" = "" ]
 then
 	WISH=$DEFAULT_WISH
@@ -122,7 +126,8 @@
 
 echo "Enter the path to the Magnus temporary directory:"
 echo -n  "[$DEFAULT_MAGNUSTMP]"
-read MAGNUSTMP
+# read MAGNUSTMP
+MAGNUSTMP=
 if [ "$MAGNUSTMP" = "" ]
 then
 	MAGNUSTMP=$DEFAULT_MAGNUSTMP
--- magnus-20060324.orig/aConfigure
+++ magnus-20060324/aConfigure
@@ -19,7 +19,7 @@
 echo "Setting up default Magnus paths..."
 echo
 
-DEFAULT_MAGNUSHOME=`pwd`
+DEFAULT_MAGNUSHOME=/usr/share/magnus
 DEFAULT_MAGNUSCURRENTHOME=`pwd`
 DEFAULT_WISH=/usr/bin/wish
 DEFAULT_MAGNUSTMP=/tmp
@@ -28,7 +28,8 @@
 echo "Enter the path to the Magnus source tree (the directory where"
 echo "the Magnus source is currently located):"
 echo -n "[$DEFAULT_MAGNUSCURRENTHOME]"
-read MAGNUSCURRENTHOME
+# read MAGNUSCURRENTHOME
+MAGNUSCURRENTHOME=
 if [ "$MAGNUSCURRENTHOME" = "" ]
 then
 	MAGNUSCURRENTHOME=$DEFAULT_MAGNUSCURRENTHOME
@@ -78,12 +79,14 @@
 
 # in the meantime, just do this
 
-MAGNUSHOME=$MAGNUSCURRENTHOME
+# MAGNUSHOME=$MAGNUSCURRENTHOME
+MAGNUSHOME=$DEFAULT_MAGNUSHOME
 
 echo
 echo "Enter the path to the Wish executable (with the addinput extensions):"
 echo -n "[$DEFAULT_WISH]"
-read WISH
+# read WISH
+WISH=
 if [ "$WISH" = "" ]
 then
 	WISH=$DEFAULT_WISH
@@ -106,7 +109,8 @@
 
 echo "Enter the path to the Magnus temporary directory:"
 echo -n  "[$DEFAULT_MAGNUSTMP]"
-read MAGNUSTMP
+# read MAGNUSTMP
+MAGNUSTMP=
 if [ "$MAGNUSTMP" = "" ]
 then
 	MAGNUSTMP=$DEFAULT_MAGNUSTMP
